File: Easy/1342-Number of Steps to Reduce a Number to Zero/Simulation.rb
# @param {Integer} num
# @return {Integer}
def number_of_steps (num)
    ret = 0

    while num != 0
        if num % 2 == 0
            num /= 2
        else
            num -= 1
        end
        ret += 1
    end

    return ret
end

File: Problemset/1342-Number of Steps to Reduce a Number to Zero/Simulation.rs
impl Solution {
    pub fn number_of_steps (num: i32) -> i32 {
        let mut num = num;
        let mut ret = 0;

        while num != 0 {
            match num % 2 {
                0 => num /= 2,
                _ => num -= 1,
            }
            ret += 1;
        }

        ret
    }
}
